bot/persistence/database_connector.py

```diff
@@ -155,6 +155,105 @@ def remove_module_role(self, role_id: int):
             db_manager.execute(queries.REMOVE_MODULE_ROLE, (role_id,))
             db_manager.commit()
 
+    def get_reaction_role(self, msg_id: int, emoji: str) -> Optional[int]:
+        """Gets the role id for the specified reaction on a specific message.
+
+        Args:
+            msg_id (int): The id of the message which has been reacted to.
+            emoji (str): The emoji of the reaction.
+
+        Returns:
+            Optional[int]: The id of the role associated with the given message + reaction.
+        """
+        with DatabaseManager(self._db_file) as db_manager:
+            result = db_manager.execute(queries.GET_REACTION_ROLE, (msg_id, emoji))
+
+            row = result.fetchone()
+            if row:
+                return int(row[0])
+            return None
+
+    def add_reaction_role(self, msg_id: int, emoji: str, role_id: int):
+        """Adds information needed for a reaction role to the table "ReactionRole".
+
+        Args:
+            msg_id (int): The id of the message which users should react to.
+            emoji (str): The emoji for the reaction role.
+            role_id (int): The id of the role for the reaction role.
+        """
+        with DatabaseManager(self._db_file) as db_manager:
+            db_manager.execute(queries.INSERT_REACTION_ROLE, (msg_id, emoji, role_id))
+            db_manager.commit()
+
+    def remove_reaction_role(self, msg_id: int, emoji: str):
+        """Removes information needed for a reaction role from the table "ReactionRole".
+
+        Args:
+            msg_id (int): The id of the message which users should react to.
+            emoji (str): The emoji for the specific reaction role.
+        """
+        with DatabaseManager(self._db_file) as db_manager:
+            db_manager.execute(queries.REMOVE_REACTION_ROLE, (msg_id, emoji))
+            db_manager.commit()
+
+    def clear_reaction_roles(self, msg_id: int) -> bool:
+        """Removes all information needed for the reaction roles of a specific message from the table "ReactionRole".
+
+        Args:
+            msg_id (int): The id of the message which reaction roles should be removed.
+
+        Returns:
+            bool: A boolean indicating if any reaction roles have been deleted.
+        """
+        with DatabaseManager(self._db_file) as db_manager:
+            affected_rows = db_manager.execute(queries.CLEAR_REACTION_ROLES, (msg_id,)).rowcount
+            db_manager.commit()
+
+            return affected_rows != 0
+
+    def add_reaction_role_group(self, msg_id: int):
```

Comment:
I don't find reaction_role_group to be a descriptive name. Maybe "make_roles_exclusive(self, msg_id)" for the method and "IsMsgExclusive" for table names etc.